### PR TITLE
feat: add settings toggle for auto lock

### PR DESCRIPTION
**Pull request by bryce-mcmath**
# Summary of Changes

Adds a settings toggle to disable the auto-lock feature

Screenshot: 
prevent_auto_lock

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/App/screens/Developer.tsx [hunk 1]
```diff
@@ -15,10 +15,10 @@ const Developer: React.FC = () => {
   const [useConnectionInviterCapability, setConnectionInviterCapability] = useState(
     !!store.preferences.useConnectionInviterCapability
   )
-  const [acceptDevCredentials, setAcceptDevCredentials] = useState<boolean>(!!store.preferences.acceptDevCredentials)
-
+  const [acceptDevCredentials, setAcceptDevCredentials] = useState(!!store.preferences.acceptDevCredentials)
```
Comment (jleach) [inline packages/legacy/core/App/screens/Developer.tsx:18]: Why `!!`?

```console
➜  ~ node
Welcome to Node.js v16.15.1.
Type ".help" for more information.
> !!true
true
> !!false
false
>
```

File: packages/legacy/core/App/screens/Developer.tsx [hunk 2]
```diff
@@ -15,10 +15,10 @@ const Developer: React.FC = () => {
...
   const [useDevVerifierTemplates, setDevVerifierTemplates] = useState(!!store.preferences.useDevVerifierTemplates)
   const [enableWalletNaming, setEnableWalletNaming] = useState(!!store.preferences.enableWalletNaming)
+  const [preventAutoLock, setPreventAutoLock] = useState(!!store.preferences.preventAutoLock)
```
Comment (jleach) [inline packages/legacy/core/App/screens/Developer.tsx:21]: This is allot of unique states. I might have condensed them into a `useReducer` like like 46 of Connection.tsx.

**Review by jleach — APPROVED** [review]
No issues with the PR but I added one comment about using useReducer rather than several independent states.

File: packages/legacy/core/App/screens/Developer.tsx (hunk 2 above)
Comment (bryce-mcmath) [reply packages/legacy/core/App/screens/Developer.tsx:21]: I will add a refactoring PR that addresses this in the future (similar to the one I did for `CommonRemoveModal`)

**Pull request merged**